### ISSUE TITLE
SelectOneMenu not creating rendered columns if it is being rendered as table for some cases 

### ISSUE THREAD

**Issue by ghost** [primefaces/primefaces#4772]
I have a <p:selectOneMenu> in which selectItem and selectItems are defined in this way: 
  <f:selectItem itemLabel="#{res.label_selectOne}" itemValue="#{null}" noSelectionOption="true"/>
  <f:selectItems value="#{caBulkUpdateDataBean.bulkUpdationList}"/>

Now after all the fetching of values and everything, 
I reached to the class SelectOneMenuRendered where this selectoneMenu goes to method encodeOptionAsTable(). 
And from here my question starts,
in this method 
we have the following code snippet:

 protected void encodeOptionsAsTable(FacesContext context, SelectOneMenu menu, List<SelectItem> selectItems, List<Column> columns)
            throws IOException 
{
//some code here
writer.startElement("tr", null);
            writer.writeAttribute("class", itemStyleClass, null);
            writer.writeAttribute("data-label", itemLabel, null);
            if (selectItem.getDescription() != null) {
                writer.writeAttribute("title", selectItem.getDescription(), null);
            }

            if (itemValue == null || (valueType != null && !valueType.isAssignableFrom(itemValue.getClass()))) {
                writer.startElement("td", null);
                writer.writeAttribute("colspan", columns.size(), null);
                writer.writeText(selectItem.getLabel(), null);
                writer.endElement("td");
            }
            else {
                for (int j = 0; j < columns.size(); j++) {
                    Column column = columns.get(j);
                    String style = column.getStyle();
                    String styleClass = column.getStyleClass();

                    writer.startElement("td", null);
                    if (style != null) {
                        writer.writeAttribute("style", style, null);
                    }
                    if (styleClass != null) {
                        writer.writeAttribute("class", styleClass, null);
                    }

                    renderChildren(context, column);
                    writer.endElement("td");
                }
            }

            writer.endElement("tr");
}

SelectOneMenuRenderer is primefaces Class and I have doubt that why 
in if-condition they have put check in form of **!valueType.isAssignableFrom(itemValue.getClass())**
because I believe that because of this check my code is not rendering columns in selectOneMenu even when value is present there, as for every Case  the given condition is giving 'false'. 
valueType and itemValue.getClass() both are of java.lang.String type. 

Earlier the given code was working in primefaces 5.3 , and there if-condition had simpple check:
`if(itemValue instanceof String) {
                writer.startElement("td", null);
                writer.writeAttribute("colspan", columns.size(), null);
                writer.writeText(selectItem.getLabel(), null);
                writer.endElement("td");
            } `

but now with the given change in condition in **PrimeFaces 7.0**, now selectOneMenu is not working as expected.

**Comment by melloware** [comment]
Please use the PrimeFaces Test project: https://github.com/primefaces/primefaces-test

To create a small reproducible sample so we can debug. 

If they are both java.lang.String the isAssignableFrom() should be "true" as here is the javadoc:

> Determines if the class or interface represented by this Class object is either the same as, or is a superclass orsuperinterface of, the class or interface represented by the specified Class parameter. It returns true if so;otherwise it returns false.

**Comment by ghost** [comment]
@melloware  Yes, isAssignableFrom() is returning "true" . 
My question is, as I will repeat again in short, 
why in selectOneMenuRenderer Class of primefaces 7.0 they have used !valueType.isAssignableFrom(itemValue.getClass()) in "if" condition which I have quoted. 
I believe it should be valueType.isAssignableFrom(itemValue.getClass())

**Comment by melloware** [comment]
@TalmeezFaizy That fix was made as part of this ticket: https://github.com/primefaces/primefaces/issues/2893

@tandraschko What do you think?

**Comment by ghost** [comment]
@melloware I went through that issue and still I am not clear why 'not' condition is there. 

`if (itemValue == null || (valueType != null && !valueType.isAssignableFrom(itemValue.getClass())))`

so here in my case,
itemValue is not null --> {itemValue==null-->false},
valueType is not null --> {valueType != null --> true}
but 
valueType.isAssignableFrom(itemValue.getClass()) is True 
-- >{!valueType.isAssignableFrom(itemValue.getClass()) --> false},
as a result, 
condition in 'if' is always coming to be 'false' for all values of 'list' passed in <selectItems....../>

Note: The list I am passing is of SelectItem type.

**Comment by tandraschko** [comment]
Please provide a cloned primefaces-test, otherwise we can't help.

**Comment by ghost** [comment]
https://github.com/TalmeezFaizy/PrimeFace-Issue-Check

Tried to link files with changes in repository, hope you can check it otherwise tell me what else to do for sharing what I am trying to do.
Really sorry for delay.

**Comment by melloware** [comment]
OK I had to make some changes in your code to get it work but attached is a runnable reproducer: 
[pf-4772.zip](https://github.com/primefaces/primefaces/files/3166236/pf-4772.zip)

Run "mvn clean jetty:run-exploded" and navigate to http://localhost:8080/primefaces-test/test.xhtml.

I see your issue and I fixed it by changing your line...
```xml
<p:selectOneMenu id="selectOptions" effect="fade" var="selOne">
```
To:
```xml
<p:selectOneMenu id="selectOptions" effect="fade">
```

The "var" signifies you want  custom content but yours are just Strings so you don't need the VAR or the custom content.  It works fine as soon as the VAR is removed.

When you use the VAR it is saying you have POJO objects and Converter and want to render them as a table.  What exactly are you trying to do??  I would say your current use case is invalid.
